### PR TITLE
[14.0][MIG] purchase_stock_price_unit_sync: Migration to 14.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 14.0

Please @sergio-teruel and @carlosdauden can you review it?

@Tecnativa TT35009

### CODE REVIEW

File: purchase_stock_price_unit_sync/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            # time being. We avoid a hard dependency as well.
+            if hasattr(line.product_id, "bom_ids") and self.env[
+                "mrp.bom"
+            ].sudo()._bom_find(
+                product=line.product_id,
+                company_id=line.company_id.id,
+                bom_type="phantom",
+            ):
+                continue
+            line.move_ids.mapped("stock_valuation_layer_ids").write(
```
Comment (carlosdauden) [inline purchase_stock_price_unit_sync/models/purchase_order.py]: Can you filter this to compatibilize with stock_landed_costs?

**Review by carlosdauden — CHANGES_REQUESTED** [review]

**Review by carlosdauden — APPROVED** [review]

**Comment by victoralmau** [comment]
Ping @sergio-teruel

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**